### DISCUSSION TITLE
3 little things (cmd, menu, icon)

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#140] [Ideas]
I'd be glad if you could look into this and add them as they would help me:

A custom right-click menu for the hover button would be valuable for me, similar to what is currently available for the close button.

"Go Back" Icon on Tab: When the icon is clicked, it will run the "go back" command to navigate back in browser history. I'd love to have it active on every tab, but do you think it can be made so that this icon is only visible when there is a site to navigate back to in history? Otherwise, the icon is hidden.

Toggle Wrap Command: Usually, I haven't wrapped the tab title, but sometimes the titles are so long. Therefore, this command would be useful to activate occasionally in between.

**Comment by madprops** [comment]
I like the Go Back idea. Will add to item menu. However there's no way afaik to check if there's tab history, so it will always be visible.

Toggle Wrap sounds reasonable. I'll add it.

I don't like the second hover menu idea because there's already a hover button action on right click.

**Comment by madprops** [comment]
What I'll do instead is add 5 or 10 generic menus that can be filled and used through-out.

**Comment by madprops** [comment]
![genmenus](https://github.com/user-attachments/assets/16125a58-2cde-408f-9ec8-1ff9df2b1fe6)

**Comment by N3C2L** [comment]
Ok, thanks. Can you reproduce this: When the close button is set to be on the left side + also the hover button is set to be on the left side, the close button will not be visible in the item list?

**Comment by madprops** [comment]
Yes it's on purpose. Both can't be on the same side.

**Reply by N3C2L** [reply]
Ok hmm, but what do you think to enhance the hover button with double click + click press possibilities? Then people could use the hover button with more functions.

**Comment by madprops** [comment]
What I can do is remove the right click functionality it has right now in exchange for a second menu.

The right click action is tab selection.

Or I'll check if longpress is viable.

**Comment by madprops** [comment]
Yeah I'll add a second menu on longpress.

**Comment by N3C2L** [comment]
Great! Will use that!

**Reply by madprops** [reply]
Will also add a longpress for Close button.

**Comment by madprops** [comment]
If you disable `Hover Button Pick` it will show menu2 on rightclick as well.

**Reply by N3C2L** [reply]
I truly appreciate your kindness, thank you so much!

**Reply by madprops** [reply]
I'm gonna remove Back from the item menu. 

Because it's not an item-based menu since it's a global Go Back depending on the active tab.

So it doesn't make sense there for now.

**Reply by madprops** [reply]
> I truly appreciate your kindness, thank you so much!

Thanks. Thinking about the code makes me improve it through iteration so that's good.

**Comment by madprops** [comment]
Oh wait maybe I can.

**Reply by madprops** [reply]
Yeah i can back/forward using tab id.

**Reply by N3C2L** [reply]
I'm lost 😁 . What are you refering to?

**Reply by N3C2L** [reply]
Oh ok, I think it's this:
> I'm gonna remove Back from the item menu.
> 
> Because it's not an item-based menu since it's a global Go Back depending on the active tab.
> 
> So it doesn't make sense there for now.

**Reply by madprops** [reply]
Yeah I'm gonna remove it from the item menu for now because I can't control its visibility and it's confusing when it is available and there's no back history. You can add it to the hover button etc.

**Reply by N3C2L** [reply]
You're probably right about that. The item list would indeed look very cluttered if there were things that had no function when clicked. Thanks anyway for trying. And thank you in any case for the other implementations.
